### PR TITLE
RavenDB-22038 - ExecutePutSubscriptionBatch is producing a lot of tasks during restore

### PR DESCRIPTION
**Pull request by grisha-kotler**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-22038/ExecutePutSubscriptionBatch-is-producing-a-lot-of-tasks-during-restore

### Additional description

Execute OnDatabaseChanges per each database and not per each subscription

### Type of change

- [ ] Bug fix
- [ ] Regression bug fix
- [x] Optimization
- [ ] New feature

### How risky is the change?

- [x] Low 
- [ ] Moderate 
- [ ] High
- [ ] Not relevant

### Backward compatibility

- [x] Non breaking change
- [ ] Ensured. Please explain how has it been implemented?
- [ ] Breaking change
- [ ] Not relevant

### Is it platform specific issue?

- [ ] Yes. Please list the affected platforms.
- [x] No

### Documentation update

- [ ] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [x] No documentation update is needed 

### Testing by Contributor

- [ ] Tests have been added that prove the fix is effective or that the feature works
- [ ] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [x] It has been verified by manual testing

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [x] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [x] No

### UI work

- [ ] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [x] No UI work is needed

### CODE REVIEW

**Review by ppekrol — APPROVED** [review]

File: src/Raven.Server/ServerWide/ClusterStateMachine.cs [hunk 1]
```diff
@@ -730,7 +730,9 @@ private void ExecutePutSubscriptionBatch(ClusterOperationContext context, Blitta
 
             PutSubscriptionCommand updateCommand = null;
             Exception exception = null;
-            var actionsByDatabase = new Dictionary<string, List<Func<Task>>>();
+            var actions = new List<Func<Task>>();
+            var databases = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
```
Comment (ayende) [inline src/Raven.Server/ServerWide/ClusterStateMachine.cs:734]: Just use a dictionary? You have _one_ item per database, no?

Comment (grisha-kotler) [reply src/Raven.Server/ServerWide/ClusterStateMachine.cs:734]: `ExecuteManyOnDispose` is expecting to get a `List<Func<Task>>`.
From a dictionary, I can get the values as `ValueCollection` and then need to call `ToList()`.

Comment (garayx) [reply src/Raven.Server/ServerWide/ClusterStateMachine.cs:734]: wont we have just one db? this command is only created on restore of single database

Comment (ppekrol) [reply src/Raven.Server/ServerWide/ClusterStateMachine.cs:734]: @garayx Grisha is off, if you want to introduce changes, please handle them at this point via separate PR

**Pull request merged**